code/src/Pizza.ts

```diff
@@ -0,0 +1,24 @@
+import { Consumable } from './Consumable';
+
+export class Pizza extends Consumable {
+  numberOfSlices: number;
+  slicesEaten: number = 0;
+  constructor(numberOfSlices: number, spoiled: boolean) {
+    super('Pizza', 0, 0, spoiled);
+    this.numberOfSlices = numberOfSlices;
+  }
+
+  eat(): string {
+    if (this.slicesEaten < this.numberOfSlices) {
+      this.slicesEaten++;
+
+      if (this.slicesEaten >= this.numberOfSlices) {
+        this.consumed = true;
+      }
+      if (this.spoiled) return `You eat the ${this.name}.`;
```

Comment:
From my understanding of the task, you should only have logic with slices of pizza here; logic regarding spoiled and consumed should be defined in "use" method of the parent (Consumable) class, so you can call Pizza.use() and get all the logic + custom implementation of "eat" in Pizza class.